tools/rtpsrelay/ReaderListener.h

```diff
@@ -0,0 +1,29 @@
+#ifndef RTPSRELAY_READER_LISTENER_H_
+#define RTPSRELAY_READER_LISTENER_H_
+
+#include "AssociationTable.h"
+#include "RelayTypeSupportImpl.h"
+
+class ReaderListener : public DDS::DataReaderListener {
```

Comment:
Could the Writer & Reader Listeners be combined to a single class, and also the Publication and Subscription Listeners?  One of them would listen for updates from peer relays, the other from the BITs.